multichain-api-gateway/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+use std::{collections::HashMap, net::TcpListener, str};
+
+use actix_web::{
+    dev::Server,
+    web,
+    web::{Data, Json},
+    App, HttpRequest, HttpResponse, HttpServer, Responder,
+};
+use futures::{stream, StreamExt};
+use reqwest::Client;
+use serde_json::Value;
+use url::Url;
+
+use crate::config::{BlockScoutSettings, Instance, Settings};
+
+mod cli;
+pub mod config;
+
+#[derive(Clone)]
+pub struct APIsEndpoints {
+    apis: Vec<(Instance, Url)>,
+    concurrent_requests: usize,
+}
+
+/// Assumptions:
+/// 1. All api calls expected to have trailing path "/api"
+///     e.g. "<base_url>/<...>/<...>/api?<query>"
+/// 2. First two segments of path of api call expected to be (network, chain)
+///     e.g. "<base_url>/<network>/<chain>/<...>/<...>"
+/// Taking it to account, we expect the following api call urls:
+///     e.g. <base_url>/<network>/<chain>/api?<query>   
+impl TryFrom<BlockScoutSettings> for APIsEndpoints {
```

Comment:
It would be better to implement `From` here if no errors may occur during conversion.